cpg-core/src/main/java/de/fraunhofer/aisec/cpg/graph/types/Type.java

```diff
@@ -324,6 +324,8 @@ public void setAdditionalTypeKeywords(String keywords) {
   public boolean isFirstOrderType() {
     return this instanceof ObjectType
         || this instanceof UnknownType
+        || this instanceof FunctionType
+        // TODO(oxisto): convert FunctionPointerType to second order type
```

Comment:
I have no idea.

Reply:
It would essentially mean replacing the `FunctionPointerType` with a `PointerType` to a `FunctionType`, which means touching *all* the function pointer logic, which I would suggest for another PR.

Reply:
Ok, that's too much for this PR since it affects too many places (or ideally reorganize this part of the logic)